### ISSUE TITLE
sprpriority SCTRL is broken in ZSS syntax

### ISSUE THREAD

**Issue by Jesuszilla** [ikemen-engine/Ikemen-GO#1988]
### Describe the bug

sprpriority can't be used without wrapping it with an if statement, and even then, only if it's on one line

### To Reproduce

Try the following code anywhere in ZSS. It will crash I.K.E.M.E.N with an error:

```
sprpriority{ value: 2; layerno: 0; }
```

The following will also crash the engine:

```
if 1 {
    sprpriority{ value: 2; layerno: 0; }
}
```

However, this works:
```
if 1 {sprpriority{ value: 2; layerno: 0; }}
```

### Expected behavior

I should be able to write the sprpriority SCTRL normally, without having to wrap an if around it.

### Engine Version (or source code date)

Nightly

### Operating system

macOS

**Comment by Wftk** [comment]
weird, it doesnt crash for me and sctrl do work fine.

**Comment by potsmugen** [comment]
Yeah. The two examples don't crash for me either and they do change sprpriority.

However there's the fact that there's a trigger and a state controller with the exact same name. Might cause issues somewhere. And it's not like `Helper` either where you can see what comes next to tell if it's a trigger or sctrl.

**Comment by Jesuszilla** [comment]
Hmm. Maybe the location does matter? I tried -2 and I think -3 if that matters. Doing it in state 0, I apparently get no "Nothing assigned" error.

**Comment by K4thos** [comment]
pasted to kfmz negative states - can't reproduce.

**Comment by potsmugen** [comment]
I was able to reproduce it with Jesuszilla's Z axis test KFM.

This doesn't crash:
```
[StateDef -2]
sprpriority{ value: clamp(floor(Pos Z/8), -5, 5); layerno: 0; }
```

But this does:
```
[StateDef -2]
sprpriority { value: clamp(floor(Pos Z/8), -5, 5); layerno: 0; }
```

Note the space before the brackets.

**Comment by potsmugen** [comment]
We probably ought to deprecate the current syntax in favour of something that doesn't have the same name as the sctrl. Even if it's just like `SpritePriority`. That should also be a guideline for trigger names going forward.

**Comment by rakieldev** [comment]
It looks like just changing the sctrl name fixes the issue. It seems the parser is interpreting ``sprpriority {`` with a space as a parameter and ignoring the bracket, which causes the error. I’m not sure what the best way to handle this issue would be without changing the parser

**Comment by potsmugen** [comment]
I was going to try fixing this, but then I noticed it doesn't happen in the current build anymore. Probably one of the many compiler fixes/refactors, but I have no idea which.
